payjoin/src/uri/mod.rs

```diff
@@ -111,11 +111,14 @@ impl PjUriBuilder {
         address: Address,
         origin: Url,
         #[cfg(feature = "v2")] ohttp_keys: Option<OhttpKeys>,
+        #[cfg(feature = "v2")] expiry: Option<std::time::SystemTime>,
```

Comment:
Github won't let me add a comment on line 109 but we should have a docstring for `expiry`.

Relatedly, it might be clearer if we extract these params into a custom struct for v2-only params? e.g.:

```
#[cfg(feature = "v2")]
struct V2Params {
    /// Optional OHTTP keys for v2 (only available if the "v2" feature is enabled).
    ohttp_keys: Option<OhttpKeys>,
    /// Optional expiry time for the payjoin session.
    expiry: Option<std::time::SystemTime>,
}

impl PjUriBuilder {
    pub fn new(
        address: Address,
        origin: Url,
        #[cfg(feature = "v2")] v2_params: V2Params,
    ) -> Self {
        ...
        #[cfg(feature = "v2")]
        pj.set_v2_params(v2_params);
    }
}
```

Reply:
I think this could work. Where would v2_params live?

Reply:
You mean the V2Params struct? It specifically relates to URI params so it should live in uri/mod.rs I would think.